src/reseeding.rs

```diff
@@ -83,63 +83,43 @@ impl<R: RngCore + SeedableRng, Rsdr: RngCore> ReseedingRng<R, Rsdr> {
     /// Reseed the internal PRNG.
     ///
     /// This will try to work around errors in the RNG used for reseeding
-    /// intelligently. If the error kind indicates retrying might help, it will
-    /// immediately retry a couple of times. If the error kind indicates the
-    /// seeding RNG is not ready, it will retry later, after `threshold / 256`
-    /// generated bytes. On other errors in the source RNG, this will skip
-    /// reseeding and continue using the internal PRNG, until another
-    /// `threshold` bytes have been generated (at which point it will try
-    /// reseeding again).
-    #[inline(never)]
+    /// intelligently through some combination of retrying and delaying
+    /// reseeding until later. So long as the internal PRNG doesn't fail, this
+    /// method will not fail, i.e. failures from the reseeding source are not
+    /// fatal.
     pub fn reseed(&mut self) {
-        trace!("Reseeding RNG after generating {} bytes",
-               self.threshold - self.bytes_until_reseed);
-        self.bytes_until_reseed = self.threshold;
-        let mut err_count = 0;
-        loop {
-            if let Err(e) = R::from_rng(&mut self.reseeder)
-                            .map(|result| self.rng = result) {
-                let kind = e.kind();
-                if kind.should_wait() {
-                    self.bytes_until_reseed = self.threshold >> 8;
-                    warn!("Reseeding RNG delayed for {} bytes",
-                           self.bytes_until_reseed);
-                } else if kind.should_retry() {
-                    err_count += 1;
-                    // Retry immediately for 5 times (arbitrary limit)
-                    if err_count <= 5 { continue; }
-                }
-                warn!("Reseeding RNG failed; continuing without reseeding. Error: {}", e);
-            }
-            break; // Successfully reseeded, delayed, or given up.
-        }
+        // Behaviour is identical to `try_reseed`; we just squelch the error.
+        let _res = self.try_reseed();
     }
 
     /// Reseed the internal RNG if the number of bytes that have been
     /// generated exceed the threshold.
     ///
     /// If reseeding fails, return an error with the original cause. Note that
-    /// if the cause has a permanent failure, we report a transient error and
-    /// skip reseeding; this means that only two error kinds can be reported
-    /// from this method: `ErrorKind::Transient` and `ErrorKind::NotReady`.
+    /// in case of error we simply delay reseeding, allowing the generator to
+    /// continue its output of random data and try reseeding again later;
+    /// because of this we always return kind `ErrorKind::Transient`.
```

Comment:
Okay, I think keeping `Transient` is a good idea. So basically we only changed `ErrorKind::Other` to some other name?